esti/hooks_test.go

```diff
@@ -42,6 +43,29 @@ func TestHooksSuccess(t *testing.T) {
 	})
 }
 
+func waitForListRepositoryRunsLen(ctx context.Context, t *testing.T, repo, ref string, l int) *api.ActionRunList {
```

Comment:
If we extract the exponential back off configured in this function to another function - the test code just needs to use it and pass, and we will get assert failures to point to the right error during the test execution.